src/smt/theory_str_noodler/theory_str_noodler.cpp

```diff
@@ -935,7 +935,11 @@ namespace smt::noodler {
                 // we did not find a solution (with satisfiable length constraints)
                 // we need to block current assignment
                 STRACE("str", tout << "assignment unsat " << mk_pp(block_len, m) << std::endl;);
-                block_curr_len(block_len);
+                if(m.is_false(block_len)) {
+                    block_curr_len(block_len, false, true);
+                } else {
+                    block_curr_len(block_len);
+                }
```

Comment:
What is this for? Why do we need to handle `false` case differently?

Reply:
In that case we get the unsat core `false` and if we add this as a theory lemma, we get unsat immediately